hexa/connections/dhis2_client_helper.py

```diff
@@ -0,0 +1,46 @@
+from openhexa.toolbox.dhis2 import DHIS2
+
+from hexa.user_management.models import User
+from hexa.workspaces.models import Connection
+
+
+def get_client_by_slug(slug: str, user: User) -> DHIS2 | None:
+    """
+    Gets DHIS2 client from workspace by its slug
+    """
+    try:
+        dhis2_connection = (
+            Connection.objects.filter_for_user(user).filter(slug=slug).first()
+        )
+    except Connection.DoesNotExist:
+        return None
+
+    if not dhis2_connection:
+        return None
+
+    fields = dhis2_connection.fields.all()
+    dhis_connection_dict = {field.code: field.value for field in fields}
+
+    return DHIS2(
+        url=dhis_connection_dict.get("url"),
+        username=dhis_connection_dict.get("username"),
+        password=dhis_connection_dict.get("password"),
+    )
+
+
+def query_dhis2_metadata(dhis2: DHIS2, type: str, **kwargs) -> dict:
+    """
+    Gets metadata from DHIS2
+    """
+    kwargs.setdefault("fields", "id,name")
```

Comment:
[Minor] since we enforce `id` and `name` for the result in 
```
@dhis2_connection.field("query")
def resolve_query(dhis2_client, info, **kwargs):
```
I would enforce the `fields` to `id,name` at the same place as well and keep the `query_dhis2_metadata` method generic as it was

Something like :

```
@dhis2_connection.field("query")
def resolve_query(dhis2_client, info, **kwargs):
    fields = ["id", "name"]
    try:
        metadata = query_dhis2_metadata(
            dhis2_client,
            type=kwargs.get("type"),
            fields = ",".join(fields),
            filter=kwargs.get("filter"),
        )

        result = [{field: item.get(field) for field in fields} for item in metadata]
```

This avoids repetition of restrained fields, collocates the restrictions and keeps the method `query_dhis2_metadata` generic for usage in other parts of the code

Reply:
I agree that coding wise it is better to do it this way and avoid repetition, but having the dictionary with explicitly set keys that match the GraphQL type does help to keep both consistent.

Reply:
Not sure I totally get your point, maybe you can explain me during the stand-up
But it's a Minor comment so I am not blocking for that